src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -4910,3 +4912,154 @@ poolinfo_valid(PG_FUNCTION_ARGS)
 							"with Citus, please contact us at "
 							"https://citusdata.com/about/contact_us")));
 }
+
+
+/*
+ * GetLocalClockValue returns logical_clock_value value stored in
+ * the Citus catalog pg_dist_local_group in given "savedValue"
+ * arg.
+ *
+ * Returns true on success, i.e.: finds a record in pg_dist_local_group,
+ * and false otherwise.
+ */
+bool
+GetLocalClockValue(uint64 *savedValue)
+{
+	ScanKey scanKey = NULL;
+	int scanKeyCount = 0;
+	bool success;
+
+	InitializeCaches();
```

Comment:
why do we need this, given that we're not modifying any fields of the metadata cache ?

Reply:
Yes, you are right that we are writing but reading from the catalog. I'm confused now, wouldn't the catalog-cache help here?

Reply:
Hmm, I guess we shouldn't need to initialize caches if we're directly reading from the catalog table.

Afaik this function is only supposed to be called when we're writing to the following object:
```c
static MetadataCacheData MetadataCache;
```

Reply:
(or to some other static hash tables/arrays etc. defined in metadata_cache.c)

Reply:
Removed it